examples/step-37/doc/intro.dox

```diff
@@ -165,100 +164,100 @@ A_\mathrm{cell} = B_\mathrm{cell}^T D_\mathrm{cell} B_\mathrm{cell},
+we first transform the vector of values on the local DoFs to a vector of
+gradients on the quadrature points. In the second loop, we multiply these
+gradients by the integration weight. The third loop applies the second
+gradient (in transposed form), so that we get back to a vector of (Laplacian)
+values on the cell dofs.
 
 The bottleneck in the above code is the operations done by the call to
-FEValues::reinit for every <code>cell</code>, which take about as much time as the
-other steps together (at least if the mesh is unstructured; deal.II can
+FEValues::reinit for every <code>cell</code>, which take about as much time as
+the other steps together (at least if the mesh is unstructured; deal.II can
 recognize that the gradients are often unchanged on structured meshes). That
 is certainly not ideal and we would like to do better than this. What the
 reinit function does is to calculate the gradient in real space by
 transforming the gradient on the reference cell using the Jacobian of the
 transformation from real to reference cell. This is done for each basis
 function on the cell, for each quadrature point. The Jacobian does not depend
 on the basis function, but it is different on different quadrature points in
-general. If you only build the matrix once as we've done in all
-previous tutorial programs, there is nothing one can do about the need
-to call FEValues::reinit on every cell since this transformation has
-to be done when we want to compute the local matrix elements.
-
-However, in a matrix-free implementation, we are not interested in
-applying the matrix only once. Rather, in iterative solvers, we need
-to expect that we have to apply the matrix many times, and so we can
-think about whether we may be able to cache something between
-different applications. On the other hand, we realize that we must not
-cache too much data since otherwise we get back to the situation where
-memory access becomes the dominating factor.
-
-The trick is now to factor out the Jacobian transformation and first
-apply the gradient on the reference cell only. That transforms the vector of
-values on the local dofs to a vector of gradients on the quadrature
-points. There, we first apply the Jacobian that we factored out from the
-gradient, then we apply the weights of the quadrature, and we apply the
-transposed Jacobian for preparing the third loop which again uses the
-gradients on the unit cell.
+general. If you only build the matrix once as we've done in all previous
+tutorial programs, there is nothing to be optimized since FEValues::reinit
+needs to be called on every cell. In this process, the transformation is
+applied while computing the local matrix elements.
+
+In a matrix-free implementation, however, we will compute those integrals much
+more often than once. For example, iterative solvers will apply the matrix
+many times, and so we can think about whether we may be able to cache
```

Comment:
in the previous sentence you say `more often than once`. Since this sentence start with `For example`, i was expecting some number in it instead of `many times`.